### DISCUSSION TITLE
The input focus style cannot be overridden

### DISCUSSION THREAD

**Discussion by ouweiya** [tailwindlabs/tailwindcss#6037] [Help] [answered]
### What version of @tailwindcss/forms are you using?

^0.4.0-alpha.2

### What version of Node.js are you using?

v14.17.6

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### Reproduction repository

https://github.com/ouweiya/tailwindcss-forms-Dom

### Describe your issue

The input focus style should be able to be overridden, but it cannot be overridden because the css variable is incorrectly referenced.

Only by modifying the css variable can it be solved.

```css
#list [type='text']:focus {
  --tw-ring-shadow: 0 0 #0000;
}
```

**Comment by adamwathan** [comment, marked as answer]
Hey! I'm not 100% sure what you mean here — here's an example of overriding it using a utility:

https://play.tailwindcss.com/Kuxu5gxZTe

...here's an example of overriding it using `@apply`:

https://play.tailwindcss.com/BUhf3wnvGg?file=css

...and here's an example of overriding it using vanilla CSS:

https://play.tailwindcss.com/KyJraptX5H?file=css

I can't find a way that doesn't work or any evidence of a CSS variable being incorrectly referenced. Can you explain in more detail?

**Reply by ouweiya** [reply]
I found that in the tailwindcss framework, `shadow` and `ring` are two concepts, and they do not cover each other. `shadow-none` does not affect `ring`, although they are all the same property in CSS. It is my self-righteous thinking. The misunderstanding is that there is a problem with the framework. It can be seen that the author is meticulous. Everywhere is designed with great care and a lot of effort. Tailwindcss is awesome.
